lib/improver/cli/spot_extract.py

```diff
@@ -149,69 +151,148 @@ def main(argv=None):
         "required.")
 
     args = parser.parse_args(args=argv)
+    metadata_dict = load_json_or_none(args.metadata_json)
+
+    # Load Cube
     neighbour_cube = load_cube(args.neighbour_filepath)
     diagnostic_cube = load_cube(args.diagnostic_filepath)
+    lapse_rate_cube = load_cube_or_none(args.temperature_lapse_rate_filepath)
 
-    neighbour_selection_method = NeighbourSelection(
-        land_constraint=args.land_constraint,
-        minimum_dz=args.minimum_dz).neighbour_finding_method_name()
+    # Process Cube
+    result = process(neighbour_cube, diagnostic_cube, lapse_rate_cube,
+                     args.apply_lapse_rate_correction, args.land_constraint,
+                     args.minimum_dz, args.extract_percentiles,
+                     args.ecc_bounds_warning, metadata_dict,
+                     args.suppress_warnings)
+
+    # Save Cube
+    save_netcdf(result, args.output_filepath)
+
+
+def process(neighbour_cube, diagnostic_cube, lapse_rate_cube,
+            apply_lapse_rate_correction=False, land_constraint=False,
+            minimum_dz=False, extract_percentiles=None,
+            ecc_bounds_warning=False, metadata_dict=None,
+            suppress_warnings=False):
+    """Module to run spot data extraction.
+
+    Extract diagnostic data from gridded fields for spot data sites. It is
+    possible to apply a temperature lapse rate adjustment to temperature data
+    that helps to account for differences between to spot sites real altitude
```

Comment:
This sentence contains a grammatic error and a typo :)

'that helps to account for differences between to spot sites real altitude' => 'that helps to account for differences between the spot site's real altitude'